lib/logstash/outputs/scalyr.rb

```diff
@@ -991,6 +1004,46 @@ def add_client_timestamp_to_body(body)
     body[:client_timestamp] = current_time_millis.to_s
   end
 
+  # This should only be called on events that have already been determined to be too large
+  def truncate_event(event)
+    attrs = event[:attrs]
+    new_attrs = Hash.new
+    total_size = 0
+    priority_events = ["serverHost", "parser", "logfile", "severity", "message"]
+    priority_events.each do |key|
+      next unless attrs.has_key? key
+      total_size += key.bytesize
+      value = attrs.delete key
+      value_size = _get_size(value)
+      if value_size > @max_field_size_bytes
+        value.slice!(@max_field_size_bytes, value_size)
```

Comment:
Do we want to add "..." or similar to the end to signal that the field value has been truncated? (iirc, we do the same on the server side for truncated field values)